ChangeSet
  1.1530 04/03/29 17:14:42 dgibson@wocket.ozlabs.ibm.com +1 -0
  Don't oops if the iseries_veth device receives a frame destined for a vlan
  we don't know about (this is a security issue).

  drivers/net/iseries_veth.c
    1.13 04/03/29 17:14:13 dgibson@wocket.ozlabs.ibm.com +12 -0
    Don't oops if the iseries_veth device receives a frame destined for a vlan
    we don't know about (this is a security issue).

diff -Nru a/drivers/net/iseries_veth.c b/drivers/net/iseries_veth.c
--- a/drivers/net/iseries_veth.c	Tue Mar 30 00:08:17 2004
+++ b/drivers/net/iseries_veth.c	Tue Mar 30 00:08:17 2004
@@ -1265,6 +1265,18 @@
 		
 		vlan = skb->data[9];
 		dev = mFabricMgr->netdev[vlan];
+		if (! dev) {
+			veth_error("NULL device for vlan %d in veth_receive. "
+				   "DST=%02x:%02x:%02x:%02x:%02x:%02x "
+				   "SRC=%02x:%02x:%02x:%02x:%02x:%02x\n",
+				   vlan, skb->data[0], skb->data[1],
+				   skb->data[2], skb->data[3], skb->data[4],
+				   skb->data[5], skb->data[6], skb->data[7],
+				   skb->data[9], skb->data[9], skb->data[10],
+				   skb->data[11]);
+			dev_kfree_skb_irq(skb);
+			continue;
+		}
 		port = (struct veth_port *)dev->priv;
 		dest = *((u64 *) skb->data) & 0xFFFFFFFFFFFF0000;
 
.........................................................................
# vim: syntax=diff

